src/v/raft/types.cc

```diff
@@ -643,7 +644,7 @@ raft::snapshot_metadata adl<raft::snapshot_metadata>::from(iobuf_parser& in) {
 
     auto cfg = adl<raft::group_configuration>{}.from(in);
     ss::lowres_clock::time_point cluster_time{
-      adl<ss::lowres_clock::duration>{}.from(in)};
+      adl<std::chrono::milliseconds>{}.from(in)};
```

Comment:
this is safe because lowres_clock::duration had previously been chrono::milliseconds? was just concerned here that there could be some selection of underlying representation, too, and that could result in different size on the wire?

Reply:
I was concerned about that too; as far as I can tell it was milliseconds previously, and other types of `chrono::duration` are attempted to be serialized via `for_each_field`, which fails due to private members.

Reply:
i only see adl specialization in reflection/adl.h for time_point and chrono::milliseconds. but i do see an adl specialization in cluster/types.h for model::timeout_clock::duration and that is the one that is a little scary because it is using unsigned rather signed ints on the wire. do we know if that type might accidentally be selected instead of chrono::milliseconds (and if they are incompatible?)

Reply:
```c++
void adl<model::timeout_clock::duration>::to(iobuf& out, duration dur) {
    adl<std::chrono::milliseconds>{}.to(
      out, std::chrono::duration_cast<std::chrono::milliseconds>(dur));
}

model::timeout_clock::duration
adl<model::timeout_clock::duration>::from(iobuf_parser& in) {
    return std::chrono::duration_cast<duration>(
      adl<std::chrono::milliseconds>{}.from(in));
}
```

Reply:
https://github.com/redpanda-data/redpanda/pull/5051